src/test/ui/proc-macro/expr-stmt-nonterminal-tokens.stdout

```diff
@@ -0,0 +1,540 @@
+PRINT-DERIVE INPUT (DISPLAY): enum E { V = { let _ = #[allow(warnings)] 0 ; 0 }, }
+PRINT-DERIVE DEEP-RE-COLLECTED (DISPLAY): enum E { V = { let _ = #[allow(warnings)] #[allow(warnings)] 0 ; 0 }, }
```

Comment:
@Aaron1011 
The attribute in `$expr` `#[allow(warnings)] 0` is duplicated for some reason.

Reply:
The duplication reproduces with
```rust
macro mac($expr:expr) {
    print_bang!($expr);
}

fn main() {
    mac!(#[allow(warnings)] 0);
}
```
as well.

At least this is not a regression, previously these examples resulted in ICEs.

Reply:
Made #86055 for tracking this.